Fix - Wrong Value Field in AddToCart Events

rawdreeg opened the pull request:
### Changes proposed in this Pull Request:

When triggering multiple AddToCart events from a website, the value field contains a cumulative sum of all the previous AddToCart events instead of the value of the current AddToCart only.

This has a significant impact on multiple tools in the Facebook Business Manager platform.
For example, when monitoring campaign metrics the total value for AddToCart events is much higher than the actual value in WooCommerce carts.
This is because each new AddToCart event contains the cumulative sum of prices of all products in the cart:

- First AddToCart event: 1 product, value=X
- Second AddToCart event: 2 products in cart, value=X+Y
- Third AddToCart event: 3 products in cart, value=X+Y+Z

This PR fixes this.

Closes #1847.

_Replace this with a good description of your changes & reasoning._

- [ x] Do the changed files pass `phpcs` checks? Please remove `phpcs:ignore` comments in changed files and fix any issues, or delete if not practical.

### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. Check out this branch
2. Navigate to a product page, add an item to your cart and observe that the value amount in the AddToCart pixel event code matches the value of the item added to the cart.
3. Repeat this with another product and confirm the value matches the value of the item added to the cart and not the cart total.
4. Go to WP Admin -> WooCommerce -> Settings -> Products. Ensure the option “Redirect to the cart page after successful addition” is unchecked. and "Enable AJAX add to cart buttons on archives" is selected.
5. Go to the shop page and add an item to the cart, then inspect the source code (right click -> inspect) and locate the div with class "wc-facebook-pixel-event-placeholder". Confirm it contains the pixel script tag and the values are correct:

![Screenshot 2022-05-23 at 15 17 15](https://user-images.githubusercontent.com/4209011/169827991-1294205d-aec0-4c03-a4ed-a4051a91814b.jpg)
 

### Changelog entry

> Fix - Wrong Value Field in AddToCart Events

jorgemd24 reviewed (commented):
Thanks, @rawdreeg for checking this! I have tested locally and can see that `value` gets the product price however looking into the code, `content_ids` or `content_name` is sending all the items in the cart. Shouldn't this contain only the product that is being added to the cart?

![image](https://user-images.githubusercontent.com/2488994/169790046-4c07f084-6525-4f74-9758-a0c0c55b6094.png)

rawdreeg commented:
> however looking into the code, content_ids or content_name is sending all the items in the cart. Shouldn't this contain only the product that is being added to the cart?

You're right! Thanks for catching up. I updated this PR to cover the content_ids, content_name, and content properties as well. I have also updated the add_add_to_cart_event_fragment method to correctly generate the pixel script

rawdreeg commented:
@jorgemd24  this is ready for another round of reviews :-) Thanks

jorgemd24 reviewed (approved):
Thanks, @rawdreeg, it is working as expected! I see the events in the Pixel page with the right data. Something that I saw in Pixel but I think it is more related to my account or products is that the event properties, source and version, are displayed as '_removed_'.

![image](https://user-images.githubusercontent.com/2488994/169845920-e710e2f4-4a17-4ca8-8534-f6488c9d8a25.png)

rawdreeg commented:
> Thanks, @rawdreeg, it is working as expected! I see the events in the Pixel page with the right data. Something that I saw in Pixel but I think it is more related to my account or products is that the event properties, source and version, are displayed as '_removed_'.

I think FB might be hiding this in the event manager. At least according to this [article](https://www.pixelyoursite.com/custom-data-not-being-displayed). 

Thanks for reviewing!
